### PR TITLE
refactor(pairing): Extract hkdf into its own module.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
extraction from #6514

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@mozilla/fxa-devs - r?

File: app/scripts/lib/crypto/hkdf.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//import importFxaCryptoDeriver from './deriver';
+//import required from '../required';
```
Comment (philbooth) [inline app/scripts/lib/crypto/hkdf.js]: I think you meant to delete these? 😄

File: app/scripts/lib/crypto/hkdf.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * @param {Buffer} ikmBuffer - input key material
+ * @param {Buffer} saltBuffer - salt
+ * @param {Buffer} infoBuffer - info
+ * @param {Number} [length=32]
+ * @returns {Promise} resolves to output key material.
+ */
+export default (ikmBuffer, saltBuffer, infoBuffer, length = 32) => {
+  return importFxaCryptoDeriver().then(({jose}) => {
+    required(ikmBuffer, 'ikmBuffer');
+    required(saltBuffer, 'saltBuffer');
```
Comment (philbooth) [inline app/scripts/lib/crypto/hkdf.js]: Is `salt` really required? HKDF itself can work without a salt too, right?

Comment (shane-tomlinson) [reply app/scripts/lib/crypto/hkdf.js]: You are right, info is optional too. I'll make them so.

**Pull request merged**